### PR TITLE
Partial refactor of completeMembershipFromContribution

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
This is a partial of #11556 - it contains part of that code tidy up

Before
----------------------------------------
Code less maintainable

After
----------------------------------------
Code more maintainable

Technical Details
----------------------------------------
@omarabuhussein I took a look at #11556 as we were discussing and this is about the size chunk I feel comfortable reviewing / merging .... and which I feel comfortable makes sense after reviewing the code myself

It moves the loading of related membership objects to the function that @mattwire has reworked - without actually changing it to an array as Matt does or changing the processing. I took a look at the paths by which updateMembershipBasedOnCompletionOfContribution is reached and I feel pretty comfortable it is equivalent to the result of $contributionDAO->loadRelatedMembershipObjects(); in all cases - that is the main change in here. I'm not too worried about the performance impact of potentially loading that twice. I don't think it's much in the scheme of things. If you feel comfortable with this portion we can merge it & get @mattwire to rebase the larger PR

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -2441,7 +2441,7 @@ public function loadRelatedObjects(&$input, &$ids, $loadAll = FALSE) {
       }
     }
 
-    $this->loadRelatedMembershipObjects($ids);
+    $ids = $this->loadRelatedMembershipObjects($ids);
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: just ditching pass-by-param here

File: CRM/Contribute/BAO/Contribution.php [hunk 2]
```diff
@@ -3929,15 +3928,12 @@ public static function recordAdditionalPayment($contributionId, $trxnsData, $pay
...
-        // update membership details
-        if (!empty($contributionDAO->_relatedObjects['membership'])) {
-          self::updateMembershipBasedOnCompletionOfContribution(
-            $contributionDAO,
-            $contributionDAO->_relatedObjects['membership'],
-            $contributionId,
-            $trxnsData['trxn_date']
-          );
-        }
+        self::updateMembershipBasedOnCompletionOfContribution(
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: pretty much per Matt's - load relatedObjects in updateMembershipBasedOnCompletionOfContribution & do early return in there too

File: CRM/Contribute/BAO/Contribution.php [hunk 3]
```diff
@@ -4629,15 +4621,12 @@ public static function completeOrder(&$input, &$ids, $objects, $transaction, $re
...
-        self::updateMembershipBasedOnCompletionOfContribution(
-          $contribution,
-          $memberships,
-          $primaryContributionID,
-          $changeDate,
-          CRM_Core_PseudoConstant::getLabel('CRM_Contribute_BAO_Contribution', 'contribution_status_id', CRM_Utils_Array::value('contribution_status_id', $input))
-        );
-      }
+      self::updateMembershipBasedOnCompletionOfContribution(
+        $contribution,
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: no memberships param

File: CRM/Contribute/BAO/Contribution.php [hunk 4]
```diff
@@ -4598,10 +4594,6 @@ public static function completeOrder(&$input, &$ids, $objects, $transaction, $re
     self::repeatTransaction($contribution, $input, $contributionParams, $paymentProcessorId);
     $contributionParams['financial_type_id'] = $contribution->financial_type_id;
 
-    if (is_numeric($memberships)) {
```
[the remark this answers is not in the document]
Comment (eileenmcnaughton) [reply CRM/Contribute/BAO/Contribution.php]: 7 actually - $memberships not used anyway

**Review by omarabuhussein — APPROVED** [review]

**Comment by omarabuhussein** [comment]
this is good to merge from my side @eileenmcnaughton

**Comment by eileenmcnaughton** [comment]
OK - let's do this & then next round can have some tests! Merging

**Pull request merged**